src/backend/distributed/commands/index.c

```diff
@@ -921,3 +913,43 @@ DropIndexTaskList(Oid relationId, Oid indexId, DropStmt *dropStmt)
 
 	return taskList;
 }
+
+
+/*
+ * FinishConcurrentIndexCommand marks new indexes valid if they were created using
+ * the CONCURRENTLY flag, because we marked it as invalid in PostProcessIndexStmt
+ * to ensure failures leave behind an invalid index.
+ */
+void
+FinishConcurrentIndexCommand(IndexStmt *indexStmt)
+{
+	Assert(indexStmt->concurrent);
+
+	/* this logic only applies to the coordinator */
+	if (!IsCoordinator())
+	{
+		return;
```

Comment:
should we add a test in mx for that ?

Reply:
this may be a little misleading since we cannot get here if we're not on the coordinator, changing to an assert

Reply:
does it apply for below check as well ? :

```c
	if (!IsCitusTable(relationId))
	{
		return;
	}
```

Reply:
yes, changed to assert